compiler/rustc_typeck/src/impl_wf_check/min_specialization.rs

```diff
@@ -199,7 +199,7 @@ fn unconstrained_parent_impl_substs<'tcx>(
     for (predicate, _) in impl_generic_predicates.predicates.iter() {
         if let ty::PredicateKind::Projection(proj) = predicate.kind().skip_binder() {
             let projection_ty = proj.projection_ty;
-            let projected_ty = proj.ty;
+            let projected_ty = proj.term.ty();
```

Comment:
ooof specialization 🙈  . definitely needs to handle terms

Reply:
oops this also took a `TypeFoldable`, I don't know what else needs to be added here 🙃